### ISSUE TITLE
Add support for Mongos topology

### ISSUE THREAD

**Issue by kapetan** [mafintosh/mongojs#241]
`mongodb-core` also supports the [Mongos topology](https://github.com/christkv/mongodb-core/blob/f600b7449bcf2707a9b84fd1c5c14f5236019941/index.js#L5), which is not exposed by `mongojs`. 

Maybe `lib/get-topology.js` should also be a able to return an `Mongos` instance.

**Comment by saintedlama** [comment]
Implemented a very experimental version in branch saintedlama/mongos https://github.com/mafintosh/mongojs/tree/saintedlama/mongos would be awesome if you could give that version a try (= alpha test)

**Comment by kapetan** [comment]
That was fast. I gave it a try, and I can connect and retrieve documents, but it throws a type error when trying to close the database.

The script I used.

```js
var mongojs = require('mongojs');

var connection = '...';
var options = {
	mongos: true,
	ssl: true,
	sslValidate: false,
	rejectUnauthorized: false,
	authMechanism: 'ScramSHA1'
};

var db = mongojs(connection, ['appUsers'], options);

db.appUsers.findOne({}, function(err, user) {
	console.log(user); // Returns the user
	db.close(); // Call to close throws a TypeError
});
```

The stack of the error.

```
/private/tmp/mj/node_modules/mongojs/node_modules/mongodb-core/lib/cursor.js:247
      throw err;
            ^
TypeError: Object [object Object] has no method 'destroy'
    at /private/tmp/mj/node_modules/mongojs/lib/database.js:66:12
    at apply (/private/tmp/mj/node_modules/mongojs/node_modules/thunky/index.js:16:28)
    at null._getServer (/private/tmp/mj/node_modules/mongojs/node_modules/thunky/index.js:27:3)
    at Database.close (/private/tmp/mj/node_modules/mongojs/lib/database.js:64:8)
    at /private/tmp/mj/index.js:15:5
    at /private/tmp/mj/node_modules/mongojs/lib/collection.js:51:5
    at handleCallback (/private/tmp/mj/node_modules/mongojs/node_modules/mongodb-core/lib/cursor.js:244:5)
    at nextFunction (/private/tmp/mj/node_modules/mongojs/node_modules/mongodb-core/lib/cursor.js:739:5)
    at /private/tmp/mj/node_modules/mongojs/node_modules/mongodb-core/lib/cursor.js:664:7
    at queryCallback (/private/tmp/mj/node_modules/mongojs/node_modules/mongodb-core/lib/cursor.js:321:5)
```

**Comment by saintedlama** [comment]
Could you give it a try with a non auth mongos connection. I guess this could narrow issue issue #224 - I suspect the error to arise only in auth environments

**Comment by saintedlama** [comment]
Any update?

**Comment by kapetan** [comment]
Sorry I didn't answer sooner, but I didn't have a mongos setup without authentication running, and I didn't have the time to try to set it up locally. I'll see if I can do it one of these days.

**Comment by saintedlama** [comment]
:+1: great thanks for your help

**Comment by saintedlama** [comment]
Should be fixed in version 2.0 (released just some seconds ago)
